app/src/main/java/com/travelguide/activities/TravelGuideActivity.java

```diff
@@ -365,24 +406,29 @@ public void onAnimationUpdate(ValueAnimator valueAnimator) {
         }
     }
 
-    private void setHeaderProfileInfo() {
-        final ParseUser currentUser = ParseUser.getCurrentUser();
-
-        if (currentUser != null) {
-            try {
-                ParseFile parseFile = currentUser.getParseFile("profileThumb");
-                byte[] data = parseFile.getData();
-                Bitmap bitmap = BitmapFactory.decodeByteArray(data, 0, data.length);
-                ivProfile.setImageBitmap(bitmap);
-            } catch (Exception e) {
-                e.printStackTrace();
-            }
+    private void setHeaderProfileInfo(boolean force) {
+        if (force
+                || loginStatus != Preferences.readBoolean(this, Preferences.User.LOG_IN_STATUS)
+                || (loginStatus && TextUtils.isEmpty(tvProfileUsername.getText()))
+                || (!loginStatus && !TextUtils.isEmpty(tvProfileUsername.getText()))) {
+            final ParseUser currentUser = ParseUser.getCurrentUser();
+            if (currentUser != null) {
+                try {
+                    ParseFile parseFile = currentUser.getParseFile("profileThumb");
+                    byte[] data = parseFile.getData();
+                    Bitmap bitmap = BitmapFactory.decodeByteArray(data, 0, data.length);
+                    ivProfile.setImageBitmap(bitmap);
+                } catch (Exception e) {
+                    e.printStackTrace();
+                }
 
-            tvProfileUsername.setText(currentUser.getUsername());
-            tvProfileEmail.setText(currentUser.getEmail());
-        } else {
-            tvProfileUsername.setText("");
-            tvProfileEmail.setText("");
+                tvProfileUsername.setText(currentUser.getUsername());
+                tvProfileEmail.setText(currentUser.getEmail());
+            } else {
+                ivProfile.setImageResource(R.drawable.background_profile);
```

Comment:
This line is not right, should be `ivProfile.setImageResource(R.drawable.profile_placeholder);`